src/System Application/App/Guided Experience/src/Manual Setup/ManualSetup.Page.al

```diff
@@ -80,27 +74,59 @@ page 1875 "Manual Setup"
 
                 trigger OnAction()
                 begin
-                    if (Rec."Object Type to Run" = Rec."Object Type to Run"::Page) and (Rec."Object ID to Run" <> 0) then
-                        Page.Run(Rec."Object ID to Run");
+                    RunPage();
                 end;
             }
         }
     }
 
     var
         ManualSetupCategory: Enum "Manual Setup Category";
+        NameIndent: Integer;
         FilterSet: Boolean;
+        NameEmphasize: Boolean;
 
     trigger OnOpenPage()
     var
-        GuidedExperienceImpl: Codeunit "Guided Experience Impl.";
         GuidedExperience: Codeunit "Guided Experience";
+        GuidedExperienceImpl: Codeunit "Guided Experience Impl.";
     begin
         GuidedExperience.OnRegisterManualSetup();
-        GuidedExperienceImpl.GetContentForSetupPage(Rec, Rec."Guided Experience Type"::"Manual Setup");
+        GuidedExperienceImpl.GetContentForManualSetup(Rec);
+        Rec.SetCurrentKey("Manual Setup Category");
 
         if FilterSet then
             Rec.SetRange("Manual Setup Category", ManualSetupCategory);
+
+        if Rec.FindFirst() then; // Set selected record to first record
+    end;
+
+    trigger OnAfterGetRecord()
+    var
+        GuidedExperienceImpl: Codeunit "Guided Experience Impl.";
+    begin
+        if GuidedExperienceImpl.IsAssistedSetupSetupRecord(Rec) then
```

Comment:
Now this procedure is checking in both ManualSetup and AssistedSetup pages.

Maybe some like `IsValidObjectIDToRun` or something

Reply:
Yes, that would be a better approach. I refactored it a bit. Thanks!